### PR TITLE
[typescript-language-features] Add telemetry for package.json auto imports

### PR DESCRIPTION
**Pull request by andrewbranch**
Counterpart to https://github.com/microsoft/TypeScript/pull/38923 (and #103037)

This is my first time messing with VS Code telemetry so I have some questions and need a careful review.

With https://github.com/microsoft/TypeScript/pull/38923, we are potentially doing extra work to offer more auto-imports in completion items, so we want to validate that it’s worth it. So, this PR adds a property to the existing `completions.execute` event that says whether any of these extra items are included in the response, and adds a new event, `completions.accept`, fired upon accepting a completion item, that includes a property that says whether the accepted item was one of these extra auto-import completion items. Between these two events, we can determine how often the extra work was what the user wanted.

### CODE REVIEW

File: extensions/typescript-language-features/src/languageFeatures/completions.ts [hunk 1]
```diff
@@ -536,15 +535,47 @@ class TypeScriptCompletionItemProvider implements vscode.CompletionItemProvider<
...
 	}
 
+	private logCompletionsTelemetry(
+		duration: number,
+		response: ServerResponse.Response<Proto.CompletionInfoResponse> | undefined,
+		includesPackageJsonImport?: boolean
+	) {
+		/* __GDPR__
+			"completions.execute" : {
+				"duration" : { "classification": "SystemMetadata", "purpose": "FeatureInsight" },
```
Comment (andrewbranch) [inline extensions/typescript-language-features/src/languageFeatures/completions.ts:560]: Note: I changed the classification of these properties at @amcasey’s request.

File: extensions/typescript-language-features/src/languageFeatures/completions.ts [hunk 2]
```diff
@@ -536,15 +535,47 @@ class TypeScriptCompletionItemProvider implements vscode.CompletionItemProvider<
...
+					"${TypeScriptCommonProperties}"
+				]
+			}
+		*/
+		this.telemetryReporter.logTelemetry('completions.execute', {
+			duration: duration,
+			type: response?.type ?? 'unknown',
+			count: response?.type === 'response' && response.body ? response.body.entries.length : 0,
+			updateGraphDurationMs: response?.type === 'response' ? response.performanceData?.updateGraphDurationMs : undefined,
+			includesPackageJsonImport: includesPackageJsonImport ? 'true' : undefined,
```
Comment (andrewbranch) [inline extensions/typescript-language-features/src/languageFeatures/completions.ts:575]: Is there a convention for modeling boolean properties? Boolean values aren’t allowed in the `TelemetryProperties` type.

File: extensions/typescript-language-features/src/languageFeatures/completions.ts (hunk 1 above)
Comment (amcasey) [reply extensions/typescript-language-features/src/languageFeatures/completions.ts:560]: Generally speaking, numbers computed without consuming personal data are "SystemMetadata".

File: extensions/typescript-language-features/src/languageFeatures/completions.ts (hunk 2 above)
Comment (amcasey) [reply extensions/typescript-language-features/src/languageFeatures/completions.ts:575]: There's this: https://github.com/microsoft/vscode/blob/afdb5ec6e06bea182d7fdcde2318f391c8d55f6c/extensions/typescript-language-features/src/typescriptServiceClient.ts#L864

Comment (andrewbranch) [reply extensions/typescript-language-features/src/languageFeatures/completions.ts:575]: Which produces `'true'` or `'false'` 👍. I suspect it’s just as easy to distinguish between `'true'` and `undefined` as between `'true'` and `'false'` so I’m happy with this.

**Comment by mjbvz** [comment]
Looks good to me. Thanks!

**Pull request merged**